### DISCUSSION TITLE
FormApi setup in RVF v6: Unexpected fetcherKey and navigate props requirement

### DISCUSSION THREAD

**Discussion by carlysandler** [airjp73/rvf#388] [General]
## Environment
"@remix-run/react": "1.19.3"
"@remix-run/node": "1.19.3",
"@remix-run/serve": "1.19.3",
"@rvf/remix": "^6.1.1",
"remix-validated-form": "^4.6.8",

## Description

While migrating from remix-validated-form to RVF v6, I encountered an unexpected issue with the new FormApi setup.

## Steps to Reproduce

1. Follow the updated docs to convert legacy `ValidatedForm` to the new FormApi setup.
2. Implement the FormApi as per the documentation.
3. Run the linter on the updated code.

## Observed Behavior

The linter reports errors stating that `fetcherKey` and `navigate` are not defined in `rvfProps`.

## Expected Behavior

Based on the documentation and example code snippets, these props are not mentioned as required, nor are their use cases explained in detail.

## Workaround

To satisfy the linter, I set the props and defined them as `undefined`:

```typescript
const rvfProps = {
  // ... other props
  fetcherKey: undefined,
  navigate: undefined
};
```

## Usage

Here's a simplified version of my component showcasing the issue:

```typescript
import { FormProvider, useForm } from "@rvf/remix";
import { withZod } from "@rvf/zod";
import { zfd } from "zod-form-data";
import { z } from "zod";

type Props = {
  loginRoute: string;
};

export const otpValidator = withZod(
  zfd.formData({
    otp: zfd.text(z.string().length(6, "Invalid code")),
  })
);

const VerifyLogin = ({ loginRoute }: Props) => {
  const form = useForm({
    validator: otpValidator,
    fetcherKey: undefined,
    navigate: undefined,
    method: "post",
  });

  return (
    <FormProvider scope={form.scope()}>
      <form {...form.getFormProps()} autoComplete="off">
        {/* Form fields and buttons */}
      </form>
    </FormProvider>
  );
};

export default VerifyLogin;
```

## Questions

1. Is this a case of a mismatched peer dependency with my current version of `@remix-run/react`?
2. Should these props be documented as required in the RVF v6 documentation?
3. Can you provide more information on the use cases and importance of `fetcherKey` and `navigate` props?

## Additional Context

This issue is not apparent from the example code snippets in the documentation, which might lead to confusion for other users migrating to RVF v6.

**Comment by airjp73** [comment]
`fetcherKey` and `navigate` are directly from the options you can pass to Remix's `useSubmit` ([docs](https://remix.run/docs/en/main/hooks/use-submit#options)).

I haven't run into this issue before, and the error message isn't what I expect for a missing, required prop -- have you had any luck reproducing this?

**Reply by crestwood204** [reply]
I believe this is because we are using `"react-router-dom": "6.14.2"` in our project. `rvf` is attempting to `Pick` quite a few properties, including `navigate` and `fetcherKey` off of `SubmitOptions`, which doesn't include those properties in our version of `react-router-dom`. I have confirmed that in `react-router-dom 6.2.7`, these keys do exist on `SubmitOptions`.

Will have to see if `6.2.7` is compatible with `remix 1.19.3`.

**Reply by crestwood204** [reply]
So, I went and found the [earliest version](https://github.com/remix-run/react-router/blob/main/CHANGELOG.md#v6180) of `react-router-dom` which supports the `fetcherKey` property. Trying to set this as an override breaks our app due to conflicts in `@remix-run/react`, which I suppose should have been expected...not sure what to do though.

Uninstalling and reinstalling `remix-run/react 1.19.3` results in the same `react-router-dom` version dependency of `6.4.12`.
